common/djangoapps/third_party_auth/signals/tests/test_handlers.py

```diff
@@ -115,3 +117,205 @@ def test_saml_config_signal_handlers(
                     f"Expected '{error_message}' in error message for {description}, "
                     f"got: {error_calls[0][1][1]}"
                 )
+
+
+@ddt.ddt
+class TestSAMLProviderConfigUpdates(TestCase):
+    """
+    Test SAML provider config updates based on SAML configuration changes.
+    """
+
+    def setUp(self):
+        # Create test sites
+        self.site1 = Site.objects.get_or_create(domain='test-site1.com', name='Site 1')[0]
+        self.site2 = Site.objects.get_or_create(domain='test-site2.com', name='Site 2')[0]
+
+        # Create SAML configs (same slug, different sites)
+        self.config1 = SAMLConfigurationFactory(
+            site=self.site1,
+            slug='default',
+            entity_id='https://site1.com'
+        )
+        self.config2 = SAMLConfigurationFactory(
+            site=self.site2,
+            slug='default',
+            entity_id='https://site2.com'
+        )
+
+        # Create provider with NULL config (critical for fallback auth)
+        self.null_provider = SAMLProviderConfigFactory(
+            slug='null_provider',
```

Comment:
[Note: Some comments may not apply if you do larger test refactoring, so please start there first (see other comments)]

I made two changes below. Note that null_provider is a bit vague, and since only the variable name will be seen in the tests, it will make the tests less clear.
```suggestion
        self.provider_with_null_saml_configuration = SAMLProviderConfigFactory(
            slug='null_saml_configuration',
```